src/db.rs

```diff
@@ -230,17 +235,21 @@ impl KbsDb {
         Ok(())
     }
 
-    pub async fn get_secret_policy(&self, sec: &str) -> Result<policy::Policy> {
+    pub async fn get_secret_policy(&self, sec: &str) -> Result<Option<policy::Policy>> {
         let query_str = "SELECT polid FROM secrets WHERE secret_id = ?";
         let new_query_str = self.replace_binds(query_str);
 
         let pol_row = sqlx::query(&new_query_str)
             .bind(sec)
             .fetch_one(&self.dbpool)
             .await?;
-        let pol = pol_row.try_get::<i64, _>(0)? as u64;
-        let secret_policy = self.get_policy(pol).await?;
-        Ok(secret_policy)
+        match pol_row.try_get::<i64, _>(0) {
+            Ok(pid) => {
+                let secret_policy = self.get_policy(pid as u64).await?;
+                Ok(Some(secret_policy))
```

Comment:
Can probably combine these lines.

Reply:
I combined these statements in a single Ok(Some(...)) statement.